File: problem-002/problem2.py
# Project Euler
# Problem 2

evenTermCount = 0
evenTermSum = 0

fibNumbers = [0]

def fibonacci():
    global evenTermCount
    global evenTermSum
    global fibNumbers
    fibTerm = -1
    x = 0

    while fibTerm < 4000000:
        if x == 0:
            fibTerm = 0
        elif x == 1:
            fibTerm = 1
        elif x > 1:
            fibTerm = fibNumbers[x-1] + fibNumbers[x-2]

        fibNumbers.insert(x,fibTerm)

        if fibTerm % 2 == 0:
            evenTermCount += 1
            evenTermSum += fibTerm
        x += 1

fibonacci()
print evenTermSum

File: problem-002/problem2.js
// Project Euler
// Problem 2

function* getFibonacci() {
  let n1 = 1;
  yield n1;
  let n2 = 2;
  yield n2;

  while (true) {
    let n3 = n1 + n2;
      yield n3;
    n1 = n2;
    n2 = n3;
  }
}

const generator = getFibonacci();
let sum = 0;
let fibonacciNumber = 0;

while (fibonacciNumber < 4000000) {
  fibonacciNumber = generator.next().value;
  if (fibonacciNumber % 2 === 0)
    sum += fibonacciNumber;
}

console.log(sum);